src/visualization/plot_recipes.jl

```diff
@@ -278,20 +278,179 @@ end
+end
+
+PlotData1D(u_ode::AbstractVector, semi; kwargs...) = PlotData1D(wrap_array(u_ode, semi), semi; kwargs...)
+
+#Plot from a solution object, which is directly return when running a simulation.
+PlotData1D(sol::TrixiODESolution; kwargs...) = PlotData1D(sol.u[end], sol.prob.p; kwargs...)
+
+struct PlotMesh1D{PD<:PlotData1D}
+  plot_data::PD
+end
+
+getmesh(pd::PlotData1D) = PlotMesh1D(pd)
+
+@recipe function f(pd::PlotData1D)
+  label --> pd.variable_names[1]
+  pd.x, pd.data
+end
+
+@recipe function f(pm::PlotMesh1D)
+  @unpack plot_data = pm
+  @unpack x, mesh_vertices_x = plot_data
+
+  # Set geometric and annotation properties
+  xlims --> (x[begin], x[end])
+  legend -->  :none
+
+  # Set series properties
+  seriestype --> :vline
+  linecolor --> :grey
+  linewidth --> 1
+
+  # Return data for plotting
+  mesh_vertices_x
+end
+
+#Store multiple PlotData1D objects in one PlotDataSeries1D.
+#This is used for multi-variable equations.
+struct PlotDataSeries1D{PD<:PlotData1D}
+  plot_data::PD
+  variable_id::Int
+end
+
+Base.firstindex(pd::PlotData1D) = first(pd.variable_names)
+Base.lastindex(pd::PlotData1D) = last(pd.variable_names)
+Base.length(pd::PlotData1D) = length(pd.variable_names)
+Base.size(pd::PlotData1D) = (length(pd),)
+Base.keys(pd::PlotData1D) = tuple(pd.variable_names...)
+Base.eltype(pd::PlotData1D) = Pair{String, PlotDataSeries1D}
+function Base.iterate(pd::PlotData1D, state=1)
+  if state > length(pd)
+    return nothing
+  else
+    return (pd.variable_names[state] => pd[pd.variable_names[state]], state + 1)
+  end
 end
 
+function Base.show(io::IO, pd::PlotData1D)
+  print(io, "PlotData1D{",
+            typeof(pd.x), ",",
+            typeof(pd.data), ",",
+            typeof(pd.variable_names), ",",
+            typeof(pd.mesh_vertices_x),
+            "}(<x>, <data>, <variable_names>, <mesh_vertices_x>)")
+end
+
+function Base.getindex(pd::PlotData1D, variable_name)
+  variable_id = findfirst(isequal(variable_name), pd.variable_names)
+
+  if isnothing(variable_id)
+    throw(KeyError(variable_name))
+  end
+
+  return PlotDataSeries1D(pd, variable_id)
+end
+
+function Base.show(io::IO, pm::PlotMesh1D)
+  print(io, "PlotMesh1D{", typeof(pm.plot_data), "}(<plot_data::PlotData1D>)")
+end
+
+#Create plot from a PlotDataSeries1D object and creates subplots for each variable.
+@recipe function f(pds::PlotDataSeries1D)
+  @unpack plot_data, variable_id = pds
+  @unpack x, data, variable_names = plot_data
+
+  # Set geometric properties
+  xlims --> (x[begin], x[end])
+
+  # Set annotation properties
+  legend -->  :none
+  title --> variable_names[variable_id]
+  #colorbar --> :true
+
+  # Return data for plotting
+  x, vec(data[variable_id,:,:])
+end
+
+@recipe function f(pd::PlotData1D)
+  # Create layout that is as square as possible, with a preference for more columns than rows if not
+  cols = ceil(Int, sqrt(length(pd)))
+  rows = ceil(Int, length(pd)/cols)
```

Comment:
I have thought about it and I don't think it makes sense to stack 3 variables as two-by-two (two plots in the first, one in the second row). Instead, I think three plots in a row are much nicer when doing this ad-hoc visualization. Maybe you can modify this here as follows (and if it works, please also update the code for the 2D plotting):
```suggestion
  if length(pd) <= 3
    cols = length(pd)
    rows = 1
  else
    cols = ceil(Int, sqrt(length(pd)))
    rows = ceil(Int, length(pd)/cols)
  end
```

Reply:
It now looks like this:

![s](https://user-images.githubusercontent.com/72009492/109501231-9a299980-7a97-11eb-9242-e3fd837b2b7b.png)

Reply:
:+1: